cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -48,28 +48,18 @@ func runProfile(cmd *cobra.Command, args []string) {
 		}
 	}
 
-	if cmdUtil.IsValidProfile(profileName) {
-		err := profileActions.SetActiveProfile(profileName)
-		if err != nil {
-			atexit.ExitWithMessage(1, err.Error())
-		}
-		fmt.Printf("Profile '%s' set as active profile.\n", profileName)
-
-		err = cmdUtil.SetOcContext(profileName)
-		if err != nil {
-			if glog.V(2) {
-				fmt.Println(fmt.Sprintf("%s", err.Error()))
-				fmt.Println(fmt.Sprintf("oc testing.cli context could not changed for '%s'.", profileName))
-			}
-		}
-	} else {
-		err := profileActions.SetActiveProfile(profileName)
-		if err != nil {
-			atexit.ExitWithMessage(1, err.Error())
+	err := profileActions.SetActiveProfile(profileName)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+	fmt.Printf("Profile '%s' set as active profile.\n", profileName)
+	err = cmdUtil.SetOcContext(profileName)
```

Comment:
@LalatenduMohanty we are here capture the error but then there is no exit in case of `err ! = nil`. Also what is the use of `cmdUtil.SetOcContext` here? what would happen if we remove that part?

Reply:
Right, in some cases SetOcContext will fail and thats fine e.g when a profile is not running or newly created. However if the instance is running and `SetOcContext` fails then `developer` user will not be logged in by default which is a side effect of something wrong in the setting up the oc context. We decided to keep the error message for the verbose mode vs ignoring the error completely as it might help during debugging.

Reply:
@LalatenduMohanty I got this but I still not sure why there is any need to do `SetOcContext` for any case (even profile exist or not)? what is the side effect we have?

Reply:
Discussed with @praveenkumar over IRC and BJN